File: Data Structures & Algorithms/validate-parentheses/submission-0.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        std::unordered_map <char, char> bracketPairs  =
        {
            {')', '('},
            {']', '['},
            {'}', '{'}
        };

        for (char c: s){
            if (bracketPairs.count(c)){
                if (!stack.empty() && stack.top() == bracketPairs[c]){
                    stack.pop();
                } else {
                    return false;
                }
            } else {
                stack.push(c);
            }
        }
        return stack.empty();
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:

        count = 0

        def expand(s, l, r):
            count = 0
            while l >= 0 and r < len(s) and s[l] == s[r]:
                l -= 1
                r += 1
                count += 1
            return count

        for i in range(len(s)):
            count += expand(s, i, i)
            count += expand(s, i, i + 1)

        return count